### PR TITLE
DEPR: Remove *args and **kwargs on resample methods

### LINKED ISSUE #50977: DEPR: *args, **kwargs on resample methods
xref https://github.com/pandas-dev/pandas/pull/50913#discussion_r1086815567

Similar to https://github.com/pandas-dev/pandas/pull/50338, *args, **kwargs were deprecated and removed in 1.x for rolling-like methods, so the same should apply to resample methods as well.

### PR DESCRIPTION
**Pull request by kostyafarber**
- [ ] closes #50977 (Replace xxxx with the GitHub issue number)
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

### CODE REVIEW

**Comment by phofl** [comment]
We want to issue a deprecation warning first before removing them

**Comment by kostyafarber** [comment]
Sorry misunderstood. Do we want something like the warnings that were removed in this https://github.com/pandas-dev/pandas/pull/50338?

**Comment by phofl** [comment]
Yep the FutureWarning that was removed there

**Comment by kostyafarber** [comment]
Do I have to add tests for the warnings?

**Comment by phofl** [comment]
Yep, something with tm.assert_produces_warning

**Review by phofl — COMMENTED** [review]
I think something went wrong during merging

**Comment by kostyafarber** [comment]
What exactly?

**Comment by kostyafarber** [comment]
> What exactly? 

@phofl

**Comment by phofl** [comment]
Your diff in v2.0.0 is huge, that is odd

**Comment by kostyafarber** [comment]
Hmm it is. I may of accidentally deleted something and added it back. If I'm not getting any merge conflicts on my end then it should be okay I guess.

**Comment by phofl** [comment]
No not really, you have to revert the unnecessary changes

**Comment by kostyafarber** [comment]
Okay will look into it.

**Comment by kostyafarber** [comment]
@phofl latest commit looks okay to me

**Comment by phofl** [comment]
diff is still showing 1k lines, e.g. https://github.com/pandas-dev/pandas/pull/51042/files

**Comment by kostyafarber** [comment]
@phofl done (I hope)

File: pandas/core/resample.py [hunk 1]
```diff
@@ -2232,3 +2247,38 @@ def _asfreq_compat(index: DatetimeIndex | PeriodIndex | TimedeltaIndex, freq):
...
+    warn_args = args is not None and len(args) > 0
+    warn_kwargs = kwargs is not None and len(kwargs) > 0
+    if warn_args and warn_kwargs:
+        msg = "args and kwargs"
+    elif warn_args:
+        msg = "args"
+    elif warn_kwargs:
+        msg = "kwargs"
+    else:
+        msg = ""
```
Comment (phofl) [inline pandas/core/resample.py:2276]: can we simply return here? Avoid the additional if

**Review by phofl — COMMENTED** [review]
small comments

File: doc/source/whatsnew/v2.0.0.rst [hunk 2]
```diff
@@ -780,7 +780,7 @@ Deprecations
 - :meth:`Index.is_object` has been deprecated. Use :func:`pandas.api.types.is_object_dtype` instead (:issue:`50042`)
 - :meth:`Index.is_interval` has been deprecated. Use :func:`pandas.api.types.is_intterval_dtype` instead (:issue:`50042`)
 - Deprecated ``all`` and ``any`` reductions with ``datetime64`` and :class:`DatetimeTZDtype` dtypes, use e.g. ``(obj != pd.Timestamp(0), tz=obj.tz).all()`` instead (:issue:`34479`)
--
+- Deprecated arguments ``*args`` and ``**kwargs`` in :class:`Resampler` (:issue:`50977`)
```
[the remark this answers is not in the document]
Comment (kostyafarber) [reply doc/source/whatsnew/v2.0.0.rst:783]: Sure

File: pandas/core/resample.py (hunk 1 above)
Comment (kostyafarber) [reply pandas/core/resample.py:2276]: yeah we can

File: pandas/core/resample.py [hunk 3]
```diff
@@ -2219,3 +2234,38 @@ def _asfreq_compat(index: DatetimeIndex | PeriodIndex | TimedeltaIndex, freq):
...
+    warn_kwargs = kwargs is not None and len(kwargs) > 0
+    if warn_args and warn_kwargs:
+        msg = "args and kwargs"
+    elif warn_args:
+        msg = "args"
+    elif warn_kwargs:
+        msg = "kwargs"
+    else:
+        return
+    if msg != "":
```
Comment (phofl) [inline pandas/core/resample.py:2264]: You can remove the if now

**Review by phofl — COMMENTED** [review]
small comment, otherwise ready

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
thx @kostyafarber

**Pull request merged**